From 73155e39880a6499fa9ef913138a7d3dce5d003b Mon Sep 17 00:00:00 2001
From: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
Date: Wed, 4 Oct 2017 15:33:15 -0300
Subject: [PATCH 3/3] display-x11: Unset tiled state if _GTK_EDGE_CONSTRAINTS
 is supported

Commit c415bef5de introduced support for the new _GTK_EDGE_CONSTRAINTS
atom. If the compositor supports that atom, however, we were always
setting the tiled state, even if no actual tiling information is
available, where the correct action is to completely remove any traces
of the tiled state.

Fix that by correctly removing the tiled state when compositor supports
_GTK_EDGE_CONSTRAINTS Xatom.

https://bugzilla.gnome.org/show_bug.cgi?id=788516
---
 gdk/x11/gdkdisplay-x11.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/gdk/x11/gdkdisplay-x11.c b/gdk/x11/gdkdisplay-x11.c
index e9a263f23d..964edbccae 100644
--- a/gdk/x11/gdkdisplay-x11.c
+++ b/gdk/x11/gdkdisplay-x11.c
@@ -228,10 +228,8 @@ do_edge_constraint_state_check (GdkWindow      *window,
     }
   else
     {
-      if (!(old_state & GDK_WINDOW_STATE_TILED))
-        {
-          local_set |= GDK_WINDOW_STATE_TILED;
-        }
+      if (old_state & GDK_WINDOW_STATE_TILED)
+        local_unset |= GDK_WINDOW_STATE_TILED;
     }
 
   /* Top edge */
-- 
2.14.1

